File: frameworks/Java/kora/kora-jdbc-direct/src/main/java/io/koraframework/benchmark/repository/WorldRepository.java
package io.koraframework.benchmark.repository;

import io.koraframework.benchmark.model.Fortune;
import io.koraframework.benchmark.model.World;

import java.util.Collection;
import java.util.List;
import java.util.Set;

import ru.tinkoff.kora.database.common.UpdateCount;
import ru.tinkoff.kora.database.common.annotation.Batch;
import ru.tinkoff.kora.database.common.annotation.Query;
import ru.tinkoff.kora.database.common.annotation.Repository;
import ru.tinkoff.kora.database.jdbc.JdbcRepository;

@Repository
public interface WorldRepository extends JdbcRepository {

    @Query("SELECT id, randomnumber FROM world WHERE id = :id")
    World findById(int id);

    @Query("UPDATE world SET randomnumber = :world.randomNumber WHERE id = :world.id")
    void update(@Batch List<World> world);

    @Query("SELECT id, message FROM fortune")
    List<Fortune> fortunes();
}

File: frameworks/Java/fair/kotlin-kora1-vertx-repository-suspend/src/main/kotlin/io/techempower/benchmark/kora/repository/WorldRepository.kt
package io.techempower.benchmark.kora.repository

import io.techempower.benchmark.kora.model.Fortune
import io.techempower.benchmark.kora.model.World
import ru.tinkoff.kora.database.common.annotation.Query
import ru.tinkoff.kora.database.common.annotation.Repository
import ru.tinkoff.kora.database.vertx.VertxRepository

@Repository
interface WorldRepository : VertxRepository {

    @Query("SELECT id, randomnumber FROM world WHERE id = :id")
    suspend fun findById(id: Int): World

    @Query("SELECT randomnumber FROM world WHERE id = :id")
    suspend fun findRandomNumberById(id: Int): Int

    @Query("UPDATE world SET randomnumber = :randomNumber WHERE id = :id")
    suspend fun updateRandomNumber(id: Int, randomNumber: Int)

    @Query("SELECT id, message FROM fortune")
    suspend fun fortunes(): List<Fortune>
}
